modules/core/src/main/java/org/apache/ignite/cache/query/SqlQuery.java

```diff
@@ -130,6 +134,27 @@ public String getType() {
         return this;
     }
 
+    /**
+     * Gets the query execution timeout in milliseconds.
+     */
+    public int getTimeout() {
+        return timeout;
+    }
+
+    /**
+     * Sets the query execution timeout. Query will be automatically cancelled if the execution timeout is exceeded.
+     * @param timeout Timeout value.
+     * @param timeUnit Time granularity.
+     */
+    public void setTimeout(int timeout, TimeUnit timeUnit) {
+        A.ensure(timeUnit != TimeUnit.MICROSECONDS && timeUnit != TimeUnit.NANOSECONDS,
```

Comment:
You can create static method on `GridQueryProcessor` which will do all the needeed checks and conversions to avoid code duplication here and in `SqlFieldsQuery`.